.changeset/happy-camels-rule.md

```
@@ -0,0 +1,9 @@
+---
+'@mastra/core': minor
+---
+
+Added portable top-level string metadata predicates to advanced trace queries. Metadata keys and values are validated before storage execution and work inside recursive Boolean predicates.
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)